src/utils/DateUtils.ts

```diff
@@ -253,8 +253,33 @@ export class DateUtils {
   }
 
   private static getMomentJsFormat(format: string) {
-    // yyyy was used to format dates before implementing moment.js, which only recognizes YYYY.
-    return format.replace(/yyyy/g, 'YYYY');
+    let correctedFormat = format;
+
+    const fourLowercaseY = DateUtils.buildRegexMatchingExactCharSequence('y', 4);
+    correctedFormat = correctedFormat.replace(fourLowercaseY, 'YYYY');
+
+    const twoLowercaseY = DateUtils.buildRegexMatchingExactCharSequence('y', 2);
+    correctedFormat = correctedFormat.replace(twoLowercaseY, 'YY');
+
+    const twoLowercaseD = DateUtils.buildRegexMatchingExactCharSequence('d', 2);
+    correctedFormat = correctedFormat.replace(twoLowercaseD, 'DD');
+
+    const oneLowercaseD = DateUtils.buildRegexMatchingExactCharSequence('d', 1);
+    correctedFormat = correctedFormat.replace(oneLowercaseD, 'D');
+
+    const twoLowercaseH = DateUtils.buildRegexMatchingExactCharSequence('h', 2);
+    correctedFormat = correctedFormat.replace(twoLowercaseH, 'H');
+
+    return correctedFormat;
+  }
+
```

Comment:
Just an thought here, with a regex like this using string literals it can be a little hard to understand. Maybe it would be worth to provide an example of what the complete regex looks like instead of having the reader deconstruct it in his head?
E.g.
```
/**
  * Resulting regex with char=y and sequenceLength=2
  * /(?<!y)y{2}(?!y)/g
  */
```

Though I know it's kinda writing twice the same information 🤔

Reply:
Yeah, I prefer to rely on the code to understand `what` is happening, using comments for `why` we are doing something. The issue appears when someone changes the code without updating the comment causing it to become misleading.